transformer_engine/pytorch/ops/basic/basic_linear.py

```diff
@@ -886,12 +885,13 @@ def op_forward(
         self,
         ctx: OperationContext,
         input_: torch.Tensor,
-        prev_op: Optional[BasicOperation] = None,
-        next_op: Optional[BasicOperation] = None,
+        prev_op_grad_input_quantizer: Optional[Quantizer],
+        next_op_input_quantizer: Optional[Quantizer],
+        is_first_op: bool,
     ) -> torch.Tensor:
 
         # Check which grads are required
-        input_requires_grad = ctx.requires_grad and input_.requires_grad
```

Comment:
We want to handle the case where `input_requires_grad=False` and `weight_requires_grad=True` so that we can skip one of the backward GEMMs. I think we'll need to expand the `requires_grad` logic in the fuser with something like:
```python
for idx in basic_op_idxs:
    basic_op_ctxs[idx].requires_grad = ...
    basic_op_ctxs[idx].input_requires_grad = input_.requires_grad if idx == 0 else basic_op_ctxs[idx-1].requires_grad
```

Reply:
I see, better to keep this PR functionally equivalent to the existing code then. We'll pin this as a future optimization.